### ISSUE TITLE
Warning in gcc 12 leads to build failure for wheel

### ISSUE THREAD

**Issue by rhauser** [jborean93/pykrb5#13]
When trying to install krb5 via pip in an environment with python 3.9.12 and gcc 12 the wheel does not build.

The reason seems to be the -Werror flag and that gcc 12 has probably more warnings than older versions. But I have no idea how to modify these flags for gcc starting with the 'pip install ...' command.

```
      creating build/temp.linux-x86_64-cpython-39/src/krb5
      /cvmfs/sft.cern.ch/lcg/releases/gcc/12.1.0-57c96/x86_64-centos7/bin/gcc -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -I/cvmfs/sft-nightlies.cern.ch/lcg/nightlies/dev4/Wed/Python/3.9.12/x86_64-centos7-gcc12-opt/include/python3.9 -c src/krb5/_ccache.c -o buildtemp.linux-x86_64-cpython-39/src/krb5/_ccache.o -Werror
      src/krb5/_ccache.c: In function '__Pyx_Import':
      src/krb5/_ccache.c:24344:21: error: the comparison will always evaluate as 'true' for the pointer operand in '(char *)"krb5._ccache" + 4' must not be NULL [-Werror=address]
      24344 |             if ((1) && (strchr(__Pyx_MODULE_NAME, '.'))) {
            |                     ^~
[CTest: warning matched]       cc1: all warnings being treated as errors
      error: command '/cvmfs/sft.cern.ch/lcg/releases/gcc/12.1.0-57c96/x86_64-centos7/bin/gcc' failed with exit code 1
      [end of output]
  
[CTest: warning matched]   note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for krb5
ERROR: Could not build wheels for krb5, which is required to install pyproject.toml-based projects
```

**Comment by jborean93** [comment]
It looks like building with a freshly cythonised set of files will work with GCC 12. You can try the following to ensure it installs from the source.

```bash
pip install cython
pip install https://github.com/jborean93/pykrb5/archive/main.tar.gz
```

I'll look at pushing a new release with either cython as a build dependency or just with a new set of files to also unblock the Python 3.11 install.

**Comment by jborean93** [comment]
The changes in https://github.com/jborean93/pykrb5/pull/14 fix this problem, once released the version on PyPI will no longer contain the cythonised files meaning pip will install the latest version of Cython during the install and generate the code with whatever bugfixes they contain. In this case this has been fixed by a newer Cython version as it no longer generates code with this warning. I've tested this on Fedora rawhide which comes with GCC 12.
